### ISSUE TITLE
Running expect_column_values_to_be_unique on a Redshift column errors out

### ISSUE THREAD

**Issue by sariabod** [fivetran/great_expectations#3293]
**Describe the bug**
When running "expect_column_values_to_be_unique" validation against a Redshift  column errors out. Running the reproduced query directly on redshift has the same error.

```
sqlalchemy.exc.InternalError: (psycopg2.errors.InternalError_) Assert
DETAIL:
 -----------------------------------------------
 error: Assert
 code:   1000
 context:  col < m_num_colflds - col=3, m_num_colflds=1.
 query:   5280517
 location: tbl_trans.hpp:495
 process:  padbmaster [pid=12189]
 -----------------------------------------------

[SQL: SELECT "UnexpectedCountSubquery".unexpected_count
FROM (SELECT sum(condition) AS unexpected_count
FROM (SELECT CASE WHEN (uor_order_id IS NOT NULL AND uor_order_id IN (SELECT uor_order_id
FROM ge_tmp_b610372a GROUP BY uor_order_id
HAVING count(uor_order_id) > %(count_1)s)) THEN %(param_1)s ELSE %(param_2)s END AS condition
FROM ge_tmp_b610372a)) AS "UnexpectedCountSubquery"]
[parameters: {'count_1': 1, 'param_1': 1, 'param_2': 0}]
(Background on this error at: http://sqlalche.me/e/13/2j85)
Calculating Metrics: 33%|
```

**To Reproduce**
1. Create a table  on Redshift and populate a column with data 
2. Create an expectation for the column to be unique
3. Create a runtime batch request
4. Run validation
5. See error

**Expected behavior**
Validation should complete successfully regardless of pass or fail status

**Environment (please complete the following information):**
 - Operating System: Linux (Ubuntu 20.04)
 - Great Expectations Version:  0.13.29

**Additional context**
I disabled temp table creation and get the same error. I created a small test set with bigint, varchar, and int both with duplicate data and without. They all have the same error with the exception of bigint without duplicates. That one ran fine.

**Comment by talagluck** [comment]
Thanks for opening this issue @sariabod ! We will review internally and be in touch in the next several days.

**Comment by NathanFarmer** [comment]
@sariabod Thank you for raising this issue and I see you opened PR #3338 to address it!

**Comment by NathanFarmer** [comment]
This issue was addressed with PR #3338
